comp/core/workloadmeta/collectors/util/ecs_util.go

```diff
@@ -122,6 +122,19 @@ func ParseV4TaskContainers(
 			}
 		}
 
+		var restartCount int
+		var restartPolicy string
+		// Unless the restartPolicy is explicitly specified in the task definition,
+		// "RestartCount" will not be included in the results from the task metadata endpoint v4.
+		// Additionally, there are the following considerations.
+		// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-restart-policy.html
+		if container.RestartCount != nil {
+			restartCount = *container.RestartCount
+			// The restart policy in ECS is an object, not a string like in docker (no, always, on-failure).
+			// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_restart_policy
+			restartPolicy = "ecs"
```

Comment:
I am not sure why we end up with `ecs` while there are multiple possible values? Also is it the same container or a new container (i.e. same container id or not?)